### DISCUSSION TITLE
PackageList error cannot find symbol

### DISCUSSION THREAD

**Discussion by silencer07** [reactwg/react-native-new-architecture#229] [Q&A] [answered]
Hi,

I am trying to follow the documentation on turbo modules: https://github.com/reactwg/react-native-new-architecture/blob/main/docs/turbo-modules.md

Basically I am just copy-pasting code and running commands. But I am encountering this error on the generated code when I am trying to run the app with linked turbo module project even on dev mode

```
 > Task :app:compileDebugJavaWithJavac FAILED
/home/runner/work/codgen-error-reproducer-react-native/codgen-error-reproducer-react-native/ReproducerApp/android/app/build/generated/autolinking/src/main/java/com/facebook/react/PackageList.java:14: error: cannot find symbol
import com.rtncalculator.CalculatorPackage;
```

Reproducer github actions log: https://github.com/silencer07/codgen-error-reproducer-react-native/actions/runs/10951901375/job/30409695439#step:7:122

Few notes:
- Created a fresh RN CLI project with typescript template with the latest RN version
- Used legacy groovy gradle instead of kotlin gradle

Thanks

**Comment by cortinico** [comment, marked as answer]
Hey @silencer07,
Thanks for providing a reproducer, I was able to identify your issues:

1. As you wrote your library in Kotlin, you need to add this line 

```diff
apply plugin: 'com.android.library'
apply plugin: 'com.facebook.react'
+apply plugin: 'org.jetbrains.kotlin.android'
```

over here:
https://github.com/silencer07/codgen-error-reproducer-react-native/blob/main/RTNCalculator/android/build.gradle

2. The name of the generated Spec interface is `NativeRTNCalculatorSpec` and not `NativeCalculatorSpec` so you need to change those two lines:

```
import com.facebook.react.bridge.ReactApplicationContext
-import com.rtncalculator.NativeCalculatorSpec
+import com.rtncalculator.NativeRTNCalculatorSpec

-class CalculatorModule(reactContext: ReactApplicationContext) : NativeCalculatorSpec(reactContext) 
+class CalculatorModule(reactContext: ReactApplicationContext) : NativeRTNCalculatorSpec(reactContext) {

  override fun getName() = NAME
```

over here: https://github.com/silencer07/codgen-error-reproducer-react-native/blob/6c17bf012f41e1228df241ac6d3ecbd99f30779a/RTNCalculator/android/src/main/java/com/rtncalculator/CalculatorModule.kt#L5-L7

**Reply by silencer07** [reply]
@cortinico thanks for the help. Finally I was able to make it work.

I think this part of the wiki needs fixing. I will submit a PR to fix this

<img width="1050" alt="Screenshot 2024-09-21 at 5 32 24 PM" src="https://github.com/user-attachments/assets/3a3b2c1a-1e00-459d-81e3-de07261cdd96">

**Reply by silencer07** [reply]
Update: PR created to fix the wiki https://github.com/reactwg/react-native-new-architecture/pull/231

Thanks again @cortinico
